File: app/src/main/java/com/phlox/tvwebbrowser/utils/StringUtils.java
package com.phlox.tvwebbrowser.utils;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;

public final class StringUtils {
	public static String streamToString(InputStream stream) throws IOException {
		ByteArrayOutputStream output = new ByteArrayOutputStream();
		int read = 0;
		byte[] buffer = new byte[1024];
		while ((read = stream.read(buffer)) > -1) {
			output.write(buffer, 0, read);
		}
		String result = new String(output.toByteArray());
		output.close();
		return result;
	}
}

File: app/src/main/java/com/phlox/tvwebbrowser/utils/StringUtils.kt
package com.phlox.tvwebbrowser.utils

import java.io.ByteArrayOutputStream
import java.io.IOException
import java.io.InputStream

object StringUtils {
    @Throws(IOException::class)
    fun streamToString(stream: InputStream): String {
        val output = ByteArrayOutputStream()
        val buffer = ByteArray(1024)
        var read = stream.read(buffer)
        while (read > -1) {
            output.write(buffer, 0, read)
            read = stream.read(buffer)
        }
        val result = String(output.toByteArray())
        output.close()
        return result
    }
}
